packages/react/src/components/Grid/Grid.mdx

```diff
@@ -0,0 +1,258 @@
+build a variety of layouts. You can import these components from
+`carbon-components-react`:
+
+```js
+import { Grid, Row, Column } from 'carbon-components-react';
+```
+
+<Preview>
+  <Story id="grid--responsive-grid" />
+</Preview>
+
+## Overview
+
+Every layout starts with the `Grid` component. You can specify a `Grid` at the
+top-level of your project, or at different depths provided that it can span
+100% width of its container.
+
+Next, you will use a combination of `Row` and `Column`. You can have multiple
+`Row` components in a `Grid`, and multiple `Column` components in a `Row`. Each
+`Row` will contain all the `Column` components provided to it, as long as they
+don't span more columns than the total number of columns in the grid.
+
+To specify how many columns the `Column` component should span, you can use the
+`sm`, `md`, `lg`, `xlg`, or `max` props. These props are shorthand versions of
+the names given to each of the breakpoints defined by the
+[2x Grid](https://www.carbondesignsystem.com/guidelines/2x-grid/implementation/#responsive-options).
+In the example below, we will use the `lg` prop for the large breakpoint and the
+number `4` to specify that each `Column` component should span 4 columns at that
+breakpoint.
+
+```js
+import { Grid, Row, Column } from 'carbon-components-react';
+
+function MyComponent() {
+  return (
+    <Grid>
+      <Row>
+        <Column lg={4}>Span 4 of 12</Column>
+        <Column lg={4}>Span 4 of 12</Column>
+        <Column lg={4}>Span 4 of 12</Column>
+        <Column lg={4}>Span 4 of 12</Column>
+      </Row>
+    </Grid>
+  );
+}
+```
+
+_Note: by default, `carbon-components` ships with a 12 column grid. You can
+enable a 16 column grid, which will be the default grid in the next major
+version, by using our
+[feature flags in Sass](https://github.com/carbon-design-system/carbon/blob/master/docs/guides/sass.md#feature-flags)._
+
+You can pair up multiple breakpoint props to specify how many columns the
+`Column` component should span at different break points. In the example below,
+we will use the `sm`, `md`, and `lg` prop to specify how many columns the
+`Column` components should span at the small, medium, and large breakpoints.
+
+<Preview>
+  <Story id="grid--responsive-grid" />
+</Preview>
+
+## Grid modes
+
+There are several
+[grid modes](https://www.carbondesignsystem.com/guidelines/2x-grid/implementation/#grid-modes)
+that you can use depending on the layout effect you're looking to achieve. By
+default, `Grid` uses the wide grid mode with a 32px gutter. However, you can use
+the `narrow` or `condensed` props to enable the narrow and condensed grid modes,
+respectively.
+
+The same way that you can pass in `narrow` or `condensed` to the `Grid`
+component, you can also pass in `narrow` or `condensed` to a `Row` component to
+enable a certain grid mode but only for a row. This can be useful when you need
+to mix-and-match certain grid modes to achieve a particular layout.
+
```

Comment:
Would it make sense to add something here around this likely being the most common use of the grid? (side note, should probably update the docs on the website with a mixed example and some more explanation there as well.)